crates/viewer/re_viewer_context/src/cache/video_cache.rs

```diff
@@ -12,46 +13,42 @@ use std::sync::{
 
 // ----------------------------------------------------------------------------
 
-#[derive(Debug, PartialEq, Eq, Hash, Clone)]
-pub struct VideoCacheKey {
-    pub versioned_instance_path_hash: VersionedInstancePathHash,
-    pub media_type: Option<MediaType>,
-}
-
 struct Entry {
     used_this_frame: AtomicBool,
-    video: Option<Arc<Mutex<Video>>>,
+
+    /// Keeps failed loads around, so we can don't try again and again.
+    video: Arc<Result<Video, VideoError>>,
 }
 
-/// Caches meshes based on their [`VideoCacheKey`].
+/// Caches meshes based on media type & row id.
 #[derive(Default)]
-pub struct VideoCache(ahash::HashMap<VideoCacheKey, Entry>);
+pub struct VideoCache(ahash::HashMap<Hash64, Entry>);
 
 impl VideoCache {
+    /// Read in some video data and cache the result.
+    ///
+    /// The `row_id` should be the `RowId` of the blob.
+    /// NOTE: videos are never batched atm (they are mono-archetypes),
+    /// so we don't need the instance id here.
+    ///
+    /// TODO(andreas,jan): This effectively takes a full copy of the video data.
+    /// We should avoid this as much as possible, having pointers back to the raw
```

Comment:
We can pass in a `Blob` here instead, which is ref-counted internally

Reply:
Let's start like that here already yes 👍 
However, the problem goes a lot deeper since we pass the data on to mp4. But that moves the todo at least!